internal/ratelimit/throttle_test.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
```

Comment:
@Tulsishah  If you have ran the integration tests once and they are passing with your changes then I think we can remove this integration test as it is testing the functionality of Throttle's implementation which we have replaced with Go's rate package. After replacement, Throttle is just passing the call to Go's rate package. Hence, the integration tests are essentially testing Go's rate package.

CC' @vadlakondaswetha

Reply:
So I should remove this after adding integration tests right?

Reply:
I didnt follow the code completely, but if we have written some logic for throttling on our side, then we should definitely add unit tests.

Reply:
@sethiay was saying in throttle.go we are using functions from rate go package. So in testing we are actually testing rate package(library) functions. Correct me if I am wrong @sethiay .

Reply:
still we need to add few unit tests to check if the rate limit is working correctly as per out expectations, if its not too tough.

Reply:
added